src/custom/utils/analytics.ts

```diff
@@ -0,0 +1,20 @@
+function getDomainRegex(domain: string | undefined): RegExp | undefined {
+  return domain ? new RegExp('^' + domain.replaceAll('.', '\\.'), 'i') : undefined
```

Comment:
Basically checks that the prefix to detect the domain in which we are.

uses RegExp, cause i construct it using a string

the replaceAll, is to use the character DOT (.) instead of the . that would mean "singe character" in a regex. The double slash is because needs to be scaped.